devMethods/src/main/java/com/kamron/pogoiv/devMethods/gameMasterParser/ApplicationDatabaseUpdater.java

```diff
@@ -312,19 +312,15 @@ private static void printIntegersXml(HashMap<Integer, FormSettings> formsByPoked
                             .format(commentFormat, pokemonName);
 
                     // Devolution Number
-                    devolutionNumberFormatter.format(integerArrayFormat, dexNumberLookup.get(poke.getParentId()) - 1);
+                    String parentId = formHash.values().stream().filter(form_ -> form_.getParentId() != null)
+                            .findFirst().map(Pokemon::getParentId).orElse(null);
+                    devolutionNumberFormatter.format(integerArrayFormat, dexNumberLookup.get(parentId) - 1);
                     devolutionNumberFormatter.format(commentFormat, pokemonName);
 
                     // Evolution Candy Cost
-                    Integer evolveCandy = null;
+                    Integer evolveCandy = -1;
```

Comment:
As far as I can tell, this can now be `int`.

Reply:
ok. Still getting used to java conventions of when to use raw and when to use objects